### ISSUE TITLE
fix errors caused by ggplot2 update

### ISSUE THREAD

**Issue by jgabry** [stan-dev/rstan#222]
The plotting functions need to be fixed to work with the forthcoming update to ggplot2:

https://github.com/hadley/ggplot2/blob/master/NEWS.md

I've found a few issues already. Hopefully they're simple to fix.

**Comment by bgoodri** [comment]
@jgabry Should this be considered closed for rstan?

**Comment by jgabry** [comment]
Yeah I think so. I just went through the plotting functions using both the
current and forthcoming ggplot versions and everything seems to work ok.

**Comment by botanize** [comment]
I'm running rstan 2.8.2, ggplot2 v2.0.0 in RStudio 0.99.486 with 64bit R 3.2.3 on Mac OS X 10.11.2

I get the following error when I run `stan_plot(fit, 'mu')`, or any of the `stan_` plotting functions:

``` r
Error in if (debug) { : argument is of length zero

Enter a frame number, or 0 to exit   

 1: stan_plot(fit, "b_badge")
 2: print(p.base + xlim(xlim.use) + p.name + thm)
 3: print.ggplot(p.base + xlim(xlim.use) + p.name + thm)
 4: ggplot_gtable(data)
 5: facet_render(plot$facet, panel, plot$coordinates, theme, geom_grobs)
 6: facet_render.null(plot$facet, panel, plot$coordinates, theme, geom_grobs)
 7: coord$render_axis_h(range, theme)
 8: f(...)
 9: guide_axis(scale_details$x.major, scale_details$x.labels, "bottom", theme)
10: element_render(theme, label_render, labels, x = label_x, expand_y = TRUE)
11: ggname(paste(element, name, sep = "."), element_grob(el, ...))
12: grobName(grob, prefix)
13: element_grob(el, ...)
14: element_grob.element_text(el, ...)
15: titleGrob(label, x, y, hjust = hj, vjust = vj, angle = angle, gp = utils::modifyList(element_gp, gp), margin = margin, ex
```

I've managed to trace it to something in the theme: 
`thm <- .rstanvis_defaults$multiparam_theme`

So that `print(p.base + xlim(xlim.use) + p.name + thm)` results in the error, but `print( p.base + xlim(xlim.use) + p.name)` does not. I see that `multiparam_theme` is based on `theme_classic`, and `print(p.base + xlim(xlim.use) + p.name + theme_classic())` also works fine. Even reconstructing the entire `.rstanvis_defaults$multiparam_theme` theme from scratch works fine: 

``` r
thm2 <- theme_classic() +
  theme(axis.line = element_line(color = "#222222"),
        axis.line.y = element_line(size = .5),
        axis.line.x = element_line(size = 3),
        axis.title = element_text(face = "bold", size = 13),
        strip.background = element_blank(),
        strip.text = element_text(color = "black", face = "bold"),
        # legend.position = "bottom",
        legend.title = element_text(size = 11),
        plot.title = element_text(size = 18)) + 
  theme(axis.title= element_blank(),
        axis.text.y = element_text(face = "bold", size = 13),
        legend.position = "none",
        panel.grid.major =  element_line(size = 0.1, color = "darkgray"))

print(p.base + xlim(xlim.use) + p.name + thm2)
```

In particular it seems to be related to `axis.text.x`

I've pretty much exhausted my ability to track this down.

**Comment by bob-carpenter** [comment]
Thanks so much for going that far.

**Comment by jgabry** [comment]
Yeah, thanks for all the details. I will dig into this further.

**Comment by jrnold** [comment]
This is weird: `ggplot2` 2.0.0 added a new element, `debug`,  to `element_text, which is why you're getting the error message. For some reason none of the`element_text`objects in`.rstanvis_defaults`have the`debug`element, as well as the`margin`element. It seems to still be a theme for`ggplot2 < 2.0.0`.

``` rconsole
> rstan:::.rstanvis_defaults$theme$text
List of 8
 $ family    : chr ""
 $ face      : chr "plain"
 $ colour    : chr "black"
 $ size      : num 12
 $ hjust     : num 0.5
 $ vjust     : num 0.5
 $ angle     : num 0
 $ lineheight: num 0.9
 - attr(*, "class")= chr [1:2] "element_text" "element"
```

However, when `stan_plot_options.R` is directly sourced, `.rstanvis_defaults$theme$text` looks fine

``` rconsole
> .rstanvis_defaults$theme$text
List of 10
 $ family    : chr ""
 $ face      : chr "plain"
 $ colour    : chr "black"
 $ size      : num 12
 $ hjust     : num 0.5
 $ vjust     : num 0.5
 $ angle     : num 0
 $ lineheight: num 0.9
 $ margin    :Classes 'margin', 'unit'  atomic [1:4] 0 0 0 0
  .. ..- attr(*, "unit")= chr "pt"
  .. ..- attr(*, "valid.unit")= int 8
 $ debug     : logi FALSE
 - attr(*, "class")= chr [1:2] "element_text" "element"
```

I'm struggling to figure out what could be causing the different behaviors though.

**Comment by jgabry** [comment]
@jrnold Thanks for the assistance. Strange indeed. I'm not sure what would cause the different behaviors either.

**Comment by jrnold** [comment]
I am starting to go down the rabbit hole of how packages do Namespaces, but that rabbit hole is deep. Just a hunch, but could it have to do with depending on `ggplot2 >= 1.0.0` instead of importing or `>= 2.0.0` in the [DESCRIPTION](https://github.com/stan-dev/rstan/blob/54b1e66d5df4bd5a998cb0b8b6bda87c14b79dd9/rstan/rstan/DESCRIPTION#L19) ? Sorry, I'm debugging blind since I'm having some problems getting the rstan repo to install from source on my machine.

**Comment by jgabry** [comment]
That rabbit hole is really deep. I'm actually editing the DESCRIPTION file right this moment to require >= 2.0.0, which we were planning on doing for the next release anyway. I'm not sure if that's the source of this issue but it's certainly possible. Interestingly, I can't recreate the error using rstan 2.8.9 (current state of the develop branch) even though there haven't been any changes relevant to the ggplot stuff since the release of 2.8.2 on CRAN.

**Comment by jgabry** [comment]
This should be fixed now with the release of version 2.9.
